features/distributed/kv-store/api/src/main/java/org/opennms/features/distributed/kvstore/api/AbstractAsyncKeyValueStore.java

```diff
@@ -55,18 +54,13 @@ protected AbstractAsyncKeyValueStore(Executor executor) {
     }
 
     protected AbstractAsyncKeyValueStore() {
-        ThreadFactory threadFactory = r -> {
-            Thread t = new Thread(r, "kvstore-async-thread");
-            t.setDaemon(true);
-            return t;
-        };
         // A reasonable default executor based on available processors that degrades to synchronous processing when full
         executor = new ThreadPoolExecutor(1,
                 Runtime.getRuntime().availableProcessors() * 10,
                 60,
                 TimeUnit.SECONDS,
                 new SynchronousQueue<>(),
-                threadFactory,
+                r -> new Thread(r, "kvstore-async-thread"),
```

Comment:
Would be nice to give the threads unique names. We use Guava's  `ThreadFactoryBuilder` to help with this for other thread pools.